examples/Mixtures/slot_based.py

```diff
@@ -0,0 +1,268 @@
+
+slot1_amount = NumericalDiscreteParameter(
+    name="Slot1_Amount", values=list(np.linspace(0, 100, RESOLUTION)), tolerance=0.2
+)
+slot2_amount = NumericalDiscreteParameter(
+    name="Slot2_Amount", values=list(np.linspace(0, 100, RESOLUTION)), tolerance=0.2
+)
+slot3_amount = NumericalDiscreteParameter(
+    name="Slot3_Amount", values=list(np.linspace(0, 100, RESOLUTION)), tolerance=0.2
+)
+
+# Store all parameters.
+
+parameters = [
+    slot1_label,
+    slot2_label,
+    slot3_label,
+    slot1_amount,
+    slot2_amount,
+    slot3_amount,
+]
+
+### Constraint Setup
+
+# Like for all mixtures, let us ensure that the overall sum of slot amounts is always
+# 100.
+
+sum_constraint = DiscreteSumConstraint(
+    parameters=["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"],
+    condition=ThresholdCondition(threshold=100, operator="=", tolerance=SUM_TOLERANCE),
+)
+
+# We could have a situation where we do not care about the order of addition of
+# components to the mixture. This comes with two additional constraints.
+
+# If there is no order of addition, it does not matter whether we have two slots
+# with the same substance or just one holding the combined amounts of two slots
+# with the same slot ingredient. Thus, let us make sure that no slot contains a
+# duplicate label entry.
+
+no_duplicates_constraint = DiscreteNoLabelDuplicatesConstraint(
+    parameters=["Slot1_Label", "Slot2_Label", "Slot3_Label"]
+)
+
+# Next, we need to take care of permutation invariance. If our order of addition does
+# not matter, the result of exchanging slot 1 with slot 3 does not change the mixture,
+# i.e. the mixture slots are permutation invariant.
+
+# One complication arising for the permutation invariance in this case stems from the
+# fact that we not only have a label per slot, but also a numerical amount. Now if
+# this amount is zero, it actually does not matter what label the slot has (i.e.
+# what substance should be considered for that slot), because we are adding 0 of it to
+# the mixture anyway. In BayBE, we call this a "dependency", i.e. the slot labels
+# depend on the slot amounts and are only relevant if the amount fulfills some
+# condition (in this case "amount > 0"). The `DiscreteDependenciesConstraint` tells
+# the `DiscretePermutationInvarianceConstraint` about these dependencies so that they
+# are correctly included in the filtering process.
+
+perm_inv_constraint = DiscretePermutationInvarianceConstraint(
+    parameters=["Slot1_Label", "Slot2_Label", "Slot3_Label"],
+    dependencies=DiscreteDependenciesConstraint(
+        parameters=["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"],
+        conditions=[
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+        ],
+        affected_parameters=[["Slot1_Label"], ["Slot2_Label"], ["Slot3_Label"]],
+    ),
+)
+
+# Store all constraints.
```

Comment:
I would remove this comment as it is redundant

Reply:
removed